src/fiber_event_bus/fiber_event_bus.ml

```diff
@@ -0,0 +1,70 @@
+open Stdune
+open Fiber.O
+
+type 'a t =
+  { mutable status : [ `Open | `Closed ]
+  ; readers : [ `Closed | `Next of 'a ] Fiber.Ivar.t Queue.t
+  ; writers : ([ `Closed | `Ok ] Fiber.Ivar.t * 'a) Queue.t
+  }
+
+let to_dyn { status; readers; writers } =
+  let status_to_dyn = function
+    | `Open -> Dyn.variant "`Open" []
+    | `Closed -> Dyn.variant "`Closed" []
+  in
+  Dyn.record
+    [ "status", status_to_dyn status
+    ; "readers", Dyn.opaque readers
+    ; "writers", Dyn.opaque writers
+    ]
+;;
```

Comment:
@rgrinberg We ended up not using this in the tests. Shall I remove it together with the dep on dyn? I'm not sure how useful this really is anymore. We can always add a `Fiber_event_bus.status` function for inspecting the status if needed.